server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/management/RequestExecutor.java

```diff
@@ -0,0 +1,28 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.management;
+
+import java.util.function.Supplier;
+
+public interface RequestExecutor {
```

Comment:
there's nothing related to rabbit here. BTW, I don't see any Request in the API

Reply:
Sorry what do you mean?

Reply:
You are reinventing ExecutorService. Something that take a Callable or a Runnable and run it.
Exactly what you want.

Reply:
Yes but I do not want `shutdown` `shutdownNow` `isShutdown` `isTerminated` `awaitTermination` `<T> Future<T> submit(Runnable task, T result)` `invokeAll * 2` `invokeAny * 2`

I also find that the intent of 'retrying' is not well expressed: `ExecutorService` is intended for multithreading work, not retrying logic.